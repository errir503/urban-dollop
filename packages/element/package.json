{
	"name": "@wordpress/element",
	"version": "3.1.0",
	"description": "Element React module for WordPress.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"gutenberg",
		"element",
		"react"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/HEAD/packages/element/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git",
		"directory": "packages/element"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"engines": {
		"node": ">=12"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"types": "build-types",
	"sideEffects": false,
	"dependencies": {
		"@babel/runtime": "^7.13.10",
		"@types/react": "^16.9.0",
		"@types/react-dom": "^16.9.0",
		"@wordpress/escape-html": "file:../escape-html",
		"lodash": "^4.17.21",
		"react": "^16.13.1",
		"react-dom": "^16.13.1"
	},
	"publishConfig": {
		"access": "public"
	}
}
