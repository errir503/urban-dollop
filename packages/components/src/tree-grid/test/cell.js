/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { forwardRef } from '@wordpress/element';

/**
 * Internal dependencies
 */
import TreeGrid from '../';
import TreeGridCell from '../cell';

const TestButton = forwardRef( ( { ...props }, ref ) => (
	<button { ...props } ref={ ref }></button>
) );

describe( 'TreeGridCell', () => {
	it( 'requires TreeGrid to be declared as a parent component somewhere in the component hierarchy', () => {
		expect( () =>
			TestRenderer.create(
				<TreeGridCell>
					{ ( props ) => (
						<TestButton className="my-button" { ...props }>
							Click Me!
						</TestButton>
					) }
				</TreeGridCell>
			)
		).toThrow();
		expect( console ).toHaveErrored();
	} );

	it( 'uses a child render function to render children', () => {
		const renderer = TestRenderer.create(
			<TreeGrid>
				<tr>
					<TreeGridCell>
						{ ( props ) => (
							<TestButton className="my-button" { ...props }>
								Click Me!
							</TestButton>
						) }
					</TreeGridCell>
				</tr>
			</TreeGrid>
		);

		expect( renderer.toJSON() ).toMatchSnapshot();
	} );
} );
