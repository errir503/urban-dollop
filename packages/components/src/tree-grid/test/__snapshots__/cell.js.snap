// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeGridCell uses a child render function to render children 1`] = `
<table
  onKeyDown={[Function]}
  role="treegrid"
>
  <tbody>
    <tr>
      <td
        role="gridcell"
      >
        <button
          className="my-button"
          onFocus={[Function]}
        >
          Click Me!
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
